src/transformers/models/bit/modeling_bit.py

```diff
@@ -494,119 +522,129 @@ class BitStage(nn.Module):
 
     def __init__(
         self,
+        config,
         in_channels,
         out_channels,
         stride,
         dilation,
         depth,
         bottle_ratio=0.25,
-        groups=1,
-        avg_down=False,
-        layer_dpr=None,
-        layer_fn=BitPreActivationBottleneckLayer,
-        act_layer=None,
-        conv_layer=None,
-        norm_layer=None,
-        **layer_kwargs
+        layer_dropout=None,
     ):
         super().__init__()
 
         first_dilation = 1 if dilation in (1, 2) else 2
-        layer_kwargs = dict(act_layer=act_layer, conv_layer=conv_layer, norm_layer=norm_layer)
-        if avg_down:
-            # TODO add support for avg_down
-            raise NotImplementedError("avg_down is not implemented")
-        proj_layer = BitDownsampleConv
+
+        # Get the layer type
+        if config.layer_type == "bottleneck":
+            layer_fn = partial(BitBottleneckLayer)
+        else:
+            layer_fn = partial(BitPreActBottleneckLayer, use_activation=False)
```

Comment:
Is it normal you have a "use_activation" argument for the "PreAct" classes, which probably always uses the activation?

Reply:
Thanks! Fixed in [a3a3068](https://github.com/NielsRogge/transformers/pull/48/commits/a3a3068eeeaed6bca1f361607a7ca2ae13a98066)